### ISSUE TITLE
Install mod by URL - and allow updates of it

### ISSUE THREAD

**Issue by exetico** [ATLauncher/ATLauncher#475]
**Is your feature request related to a problem? Please describe.**
We're using a custom mod-pack, and updates it from type to type. In that cause, currently the user needs to update/add new mods manually. If not, the user needs to remove the current instance - and add it again, with the same URL. In that case, waypoints and other things are gone.

**Describe the solution you'd like**
Allow the user to add a URL as the source for the instance, and add a "Manuel update" button which will force update of the instance by fetching the data from the URL, but not overwrite local data like waypoints and so.

**Describe alternatives you've considered**
Another take could be to use the "Import" function with the URL/Local-file, and instead of forcing the user to delete the current instance of the same name, add another option called "overwrite, but keep local data" (waypoints and so) or something like that.

**Additional context**
It would really help us, with the few users who don't really understand to update and add/remove mods manually down the line. 

_Also, the Import function are currently broken, if using a URL, cause the `tmp` location arn't resolvable. I've not reported that part separatly, cause I downloaded the file instead._

**Comment by RyanTheAllmighty** [comment]
I'll have to think about the best way to accommodate this. One of the 2 solutions you mentioned should be okay.

But the real solution is to use CurseForge to host your packs, then none of this becomes an issue.

**Comment by exetico** [comment]
Hi. For our case, I don't think CurseForge would be happy, as our current pack is primary Valhelsia, but other things added on stop. In that cause, it would look like a copy of a mod-pack with minor changes.

Therefore a "update" of a instance would make sense, by providing URL or zip-file.

I hope that makes sense :-)

**Comment by Nelly1234567** [comment]
Is there potentially a Beta for this? With the release of NeoForge and its 1.20.1 compatibility with regular Forge mods, I can't upload my modpack to CurseForge (not that I really want to as the pack is made for me and some friends) due to Forge not allowing mixed mods/modloaders. Due to that, my modpack that works absolutely fine isn't accepted by Curseforge, not all the mods are available on Modrinth and would be a pain to convert now that it's made and working smoothly. 

If not, I do hope this can be worked on at a bit higher of a priority, it would be awesome to have a way to update modpacks for friends without having to have a Curseforge page or something really public.

**Comment by RyanTheAllmighty** [comment]
> Is there potentially a Beta for this? With the release of NeoForge and its 1.20.1 compatibility with regular Forge mods, I can't upload my modpack to CurseForge (not that I really want to as the pack is made for me and some friends) due to Forge not allowing mixed mods/modloaders. Due to that, my modpack that works absolutely fine isn't accepted by Curseforge, not all the mods are available on Modrinth and would be a pain to convert now that it's made and working smoothly.
> 
> If not, I do hope this can be worked on at a bit higher of a priority, it would be awesome to have a way to update modpacks for friends without having to have a Curseforge page or something really public.

There are no updates or betas for this no.

BTW there is no conversion process needed for Modrinth, simply just export the pack to Modrinth format and the launcher will do the rest, there should be nothing else to do. If you're having issues with that, feel free to pop by https://atl.pw/discord and can help you out there

But this feature is not expected to be looked at anytime soon unfortunately, as basically the only dev working on this project, my time is limited to be able to spend working on ATLauncher at the moment sorry

**Comment by CorneliusCornbread** [comment]
> I'll have to think about the best way to accommodate this. One of the 2 solutions you mentioned should be okay.
> 
> But the real solution is to use CurseForge to host your packs, then none of this becomes an issue.

The problem with hosting on CF nowadays is that many modders don't allow their mods to be redistributed via CF or Modrinth, meaning with the right collection of mods you cannot upload your modpack to _any_ Minecraft mod distribution service. This is the case I have with my modpack and many of its mods. For example Twilight Forest cannot be distributed on modrinth (and I forget which mods, CF deletes notifications that have been read, but there are modrinth mods that CF will not accept). This means we _have_ to distribute our pack ourselves.

I could host my own pack distribution service but then it would run into the same exact problem that updating from a zip does. It's not supported.

Unfortunately this was a non-issue in the past, but has no made itself an issue due to drama between mod developers are mod platforms.

**Comment by CorneliusCornbread** [comment]
Adding on to this, maybe the solution is to support packwiz? https://github.com/packwiz/packwiz

It's quite a simple format and supports self hosting, allowing modpacks that are shared privately to be distributed with ease, and technically without violating copyright as the mod itself isn't distributed.

**Comment by RyanTheAllmighty** [comment]
Yeah there's plans to support Packwiz as well as this in #496 but Packwiz is definitely not easy to work with from a complete newbie who just pass around zips, so they definitely both have their merits for sure.
